oshi-core/src/main/java/oshi/software/os/unix/solaris/SolarisFileSystem.java

```diff
@@ -182,6 +179,20 @@ public long getMaxFileDescriptors() {
         return 0L;
     }
 
+    @Override
+    public long getMaxFileDescriptorsPerProcess() {
+        final List<String> lines = FileUtil.readFile("/etc/system");
+        for (final String line : lines) {
+            if (line.matches("set.+rlim_fd_cur=\\d+")) {
```

Comment:
Please compile this regex pattern as a static variable, although this could also use a combination of `startsWith()`, `contains()`, and `getLastInt()`.

Reply:
Also this should be `rlim_fd_max`, not `rlim_fd_cur`.